airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py

```diff
@@ -71,32 +83,36 @@ def _get_stream_kwargs(self, config: ConnectorConfig):
             aws_access_key_id=role_creds.get("AccessKeyId"),
             aws_secret_access_key=role_creds.get("SecretAccessKey"),
             aws_session_token=role_creds.get("SessionToken"),
-            region=self.region,
+            region=region,
         )
         auth = AWSAuthenticator(
             token_refresh_endpoint="https://api.amazon.com/auth/o2/token",
             client_secret=config.lwa_client_secret,
             client_id=config.lwa_app_id,
             refresh_token=config.refresh_token,
-            host=self.endpoint.replace("https://", ""),
+            host=endpoint.replace("https://", ""),
         )
         stream_kwargs = {
-            "url_base": self.endpoint,
+            "url_base": endpoint,
             "authenticator": auth,
             "aws_signature": aws_signature,
             "replication_start_date": config.replication_start_date,
+            "marketplace_ids": [marketplace_id],
         }
         return stream_kwargs
 
     def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, Any]:
-        try:
-            config = ConnectorConfig.parse_obj(config)  # FIXME: this will be not need after we fix CDK
-            stream_kwargs = self._get_stream_kwargs(config)
-            merchant_listings_reports_gen = MerchantListingsReports(**stream_kwargs).read_records(sync_mode=SyncMode.full_refresh)
-            next(merchant_listings_reports_gen)
-            return True, None
-        except Exception as error:
-            return False, f"Unable to connect to Amazon Seller API with the provided credentials - {repr(error)}"
+        config = ConnectorConfig.parse_obj(config)  # FIXME: this will be not need after we fix CDK
+        stream_kwargs = self._get_stream_kwargs(config)
+
+        reports_res = requests.get(
+            url=f"{stream_kwargs['url_base']}{MerchantListingsReports.path_prefix}/reports",
+            headers={**stream_kwargs["authenticator"].get_auth_header(), "content-type": "application/json"},
+            params={"reportTypes": MerchantListingsReports.name},
+            auth=stream_kwargs["aws_signature"],
+        )
+        connected = reports_res.status_code == 200 and reports_res.json().get("payload")
+        return connected, f"Unable to connect to Amazon Seller API with the provided credentials - {reports_res.json()}"
```

Comment:
we should only return that message if there is a failure. If there is no failure the message should be empty

Reply:
Well, this erorr message is not displayed if `connected=True`. But I've found another issue, please check out the updates.